rules/file-contents.js

```diff
@@ -29,30 +29,124 @@ async function fileContents(fs, options, not = false) {
     return new Result(
       'Did not find file matching the specified patterns',
       fileList.map(f => {
-        return { passed: false, pattern: f }
+        return { passed: !options['fail-on-non-existent'], pattern: f }
       }),
       !options['fail-on-non-existent']
     )
   }
 
-  const results = await Promise.all(
-    files.map(async file => {
-      const fileContents = await fs.getFileContents(file)
-      if (!fileContents) return null
-
-      const regexp = new RegExp(options.content, options.flags)
-      const passed = fileContents.search(regexp) >= 0
-      const message = `${passed ? 'Contains' : "Doesn't contain"} ${getContent(
-        options
-      )}`
-
-      return {
-        passed: not ? !passed : passed,
-        path: file,
-        message
-      }
-    })
-  )
+  const regexp = new RegExp(options.content, options.flags)
+  let results
+
+  if (!options['display-line-numbers']) {
+    /**
+     * Default "Contains" / "Doesn't contain"
+     */
+    results = await Promise.all(
+      files.map(async file => {
+        const fileContents = await fs.getFileContents(file)
+        if (!fileContents) return null
+
+        const passed = fileContents.search(regexp) >= 0
+        const message = `${
+          passed ? 'Contains' : "Doesn't contain"
+        } ${getContent(options)}`
+
+        return {
+          passed: not ? !passed : passed,
+          path: file,
+          message
+        }
+      })
+    )
+  } else {
+    /**
+     * Add lines for each match to output.
+     */
+    results = (
+      await Promise.all(
+        files.map(async file => {
+          const fileContents = await fs.getFileContents(file)
+          if (!fileContents) return null
+
+          const envContextLength = options['context-length'] || 50
+          const split = fileContents.split(regexp)
```

Comment:
Noting for documentation that the regexp is run on the whole strong rather than each line. Does that cause any unexpected behaviour over the end of lines that is worth flagging for a user?